### PR TITLE
Add ansible-community/ansible-lint action

### PR DESCRIPTION
**Pull request by ben-alkov**
Tested in ben-alkov/ansible-role-osbs-namespace PR 1 [linter run](https://github.com/ben-alkov/ansible-role-osbs-namespace/runs/4025683626?check_suite_focus=true)

Signed-off-by: Ben Alkov <ben.alkov@redhat.com>

- CLOUDBLD-7849

### CODE REVIEW

**Comment by MartinBasti** [comment]
> > Would https://github.com/marketplace/actions/ansible-lint not work?
> 
> Probably would.
> 
> My opinion hasn't changed since we first set up this repo - I think we're better off with our own set of actions which serve our purposes well. Exceptions would be in cases where for some reason we _can't_ make something work (e.g. licensing, I guess), or setting up the action itself if overly complex, or if maintenance of a particular action would be a burden. None of those apply here. Setting up ansible-lint took me about 10 minutes. and it'll be as easy to maintain as pylint or flake8.

Isn't GH marketplace one of the huge benefits of using GH workflows, so you can just use actions from marketplace without maintaining your own?
The one from marketplace has more configuration options also.

I don't see reason why recreate the same action

**Comment by ben-alkov** [comment]
> > > Would https://github.com/marketplace/actions/ansible-lint not work?
> > 
> > 
> > Probably would.
> > My opinion hasn't changed since we first set up this repo - I think we're better off with our own set of actions which serve our purposes well. Exceptions would be in cases where for some reason we _can't_ make something work (e.g. licensing, I guess), or setting up the action itself if overly complex, or if maintenance of a particular action would be a burden. None of those apply here. Setting up ansible-lint took me about 10 minutes. and it'll be as easy to maintain as pylint or flake8.
> 
> Isn't GH marketplace one of the huge benefits of using GH workflows, so you can just use actions from marketplace without maintaining your own? The one from marketplace has more configuration options also.
> 
> I don't see reason why recreate the same action

Internal decision was made to go with custom action, based on

"""
The "official" ansible-lint action is using v4.3.7 (12 months ago); my action uses latest  v5.2.1 (13 days ago). AFAICT, we are immediately hitting [#1112][] when using ansible-lint < 4 (see here https://github.com/ben-alkov/ansible-role-osbs-namespace/actions/runs/1408598930)
"""


[#1112]: https://github.com/ansible-community/ansible-lint/issues/1112

**Review by MartinBasti — APPROVED** [review]

File: ansible-lint/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+FROM fedora:34
```
Comment (chmeliik) [inline ansible-lint/Dockerfile:1]:
```suggestion
FROM registry.fedoraproject.org/fedora:34
```

**Review by chmeliik — APPROVED** [review]
LGTM, I'd just suggest to get the Fedora image from registry.fedoraproject.org

**Pull request merged**